reflectivenotes.php

```diff
@@ -34,6 +34,10 @@
 $userid = optional_param('userid', 0, PARAM_INT);
 $instance = optional_param('instance1', 0, PARAM_INT);
 
+$usercontext = context_user::instance($userid);
+if (!has_capability('moodle/user:viewdetails', $usercontext)) {
+    throw new moodle_exception('nopermissions', 'error', '', get_string('viewuserdetails', 'role'));
+}
```

Comment:
1. Users can only add/edit their own self-reflective notes we just need to check if `$USER->id != $userid` and throw an exception if so.
2. `get_string('viewuserdetails', 'role')` -- this string doesn't exist, making the error message _Sorry, but you do not currently have permissions to do that ([[viewuserdetails]])._
3. We should probably ensure the items students are leaving feedback for are actually items on courses they're enrolled on, maybe that should be a separate story.

Reply:
Added user verification and message, let me know if it's ok